util.c

```diff
@@ -394,3 +394,50 @@ void handleRMCallError(RedisModuleCtx *ctx, int ret_errno, const char *cmd, size
     RedisModule_ReplyWithError(ctx, errmsg);
     RedisModule_Free(errmsg);
 }
+
+void FillShardSlots(RedisRaftCtx *rr, ShardGroup *sg)
+{
+    char *str = RedisModule_Strdup(rr->config->slot_config);
+
+    sg->slot_ranges_num = 1;
+    char *pos = str;
+    while ((pos = strchr(pos+1, ','))) {
+        sg->slot_ranges_num++;
+    }
+    sg->slot_ranges = RedisModule_Calloc(sg->slot_ranges_num, sizeof(ShardGroupSlotRange));
+
+    char *saveptr = NULL;
+    char *token = strtok_r(str, ",", &saveptr);
+    for (int i = 0; i < sg->slot_ranges_num; i++) {
+        unsigned long val;
+        if ((pos = strchr(token, ':'))) {
+            *pos = '\0';
+            val = strtoul(token, NULL, 10);
+            sg->slot_ranges[i].start_slot = val;
```

Comment:
Maybe add a function comment that indicates we assume input slot config is valid?